src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructure.java

```diff
@@ -466,56 +513,138 @@ public void loadStructure(@NotNull final String name, int rotateTimes, BlockPos
                 ((AbstractJobStructure) job).setStructure(null);
             }
 
-            ((AbstractJobStructure) job).getStructure().rotate(rotateTimes);
+            ((AbstractJobStructure) job).getStructure().rotate(rotateTimes, world, position, isMirrored ? Mirror.FRONT_BACK : Mirror.NONE);
             ((AbstractJobStructure) job).getStructure().setPosition(position);
         }
     }
 
+    /**
+     * Check if the structure tusk has been canceled.
+     * @return true if reset to idle.
+     */
+    protected abstract boolean checkIfCanceled();
+
+    /**
+     * Iterates through all the required resources and stores them in the building.
+     */
     private void requestMaterials()
     {
-        //We need to deal with materials
-        if (!Configurations.builderInfiniteResources && job instanceof JobBuilder)
+        final JobBuilder builderJob = (JobBuilder) job;
+        while (builderJob.getStructure().findNextBlock())
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Reduce the total number of break and continue statements in this loop to use at most one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS135)